instrumentation/http/src/test/java/brave/http/HttpRequestMatchersTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2013-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.http;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static brave.http.HttpRequestMatchers.methodEquals;
+import static brave.http.HttpRequestMatchers.pathStartsWith;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HttpRequestMatchersTest {
+  @Mock HttpRequest httpRequest;
+
+  @Test public void methodEquals_matched() {
+    when(httpRequest.method()).thenReturn("GET");
```

Comment:
Might want to add a test that matches against lowercase, to explicitly document it doesn't match (or alternatively change the implementation to case insensitive).

Reply:
interesting.. haven't run across mixed case method names. I'll document as such

Reply:
I'm generally nervous on whether the framework I'm using happens to return all caps or lowercase for a method that returns http method. I think Armeria tags (or tagged? not sure of the new parser changed it) lowercase so I might actually assume lowercase here.

Reply:
tags sure but this is the actual http request type, and I've literally never seen that despite an inordinate amount of libraries. Never had any issues raised in a decade over mixed case http method. I'm not even sure http spec would allow it..

are you sure we should do mixed case? it would only be here as I suspect all over the internet http methods are presumed caps..

Reply:
yeah it is case sensitive. https://tools.ietf.org/html/rfc7231#section-4.1

I think you probably were thinking about tag converters which do often downcase things.

Reply:
@anuraaga this is not a function of tags, it is a function of a request object. Also this logic is just carried over, as since the beginning of the http abstraction it has been uppercase. That means however many 10 libraries haven't reported an issue in years.

I will mark HttpRequest explicitly that it must follow RFC, so if for some weird reason armeria decides to lowercase something it should uppercase it in the adapter. It hasn't so far coerced to downcase else our tests would have failed anyway. Remember we have ITs on conversion of http.method.

Reply:
I'm good if it's documented - for me it's just that since I've seen both `GET` and `get` in different places, I'd always need either a doc or code to be absolutely sure which one I'm getting from a library.

Reply:
cool I added a commit to clarify. cheers